azurerm/internal/services/media/tests/media_streaming_endpoint_resource_test.go

```diff
@@ -0,0 +1,196 @@
+		CheckDestroy: testCheckAzureRMStreamingEndpointDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMStreamingEndpoint_CDN(data),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(data.ResourceName, "cdn_profile", "MyCDNProfile"),
+					resource.TestCheckResourceAttr(data.ResourceName, "cdn_provider", "StandardVerizon"),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMStreamingEndpoint_MaxCacheAge(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_media_streaming_endpoint", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMStreamingEndpointDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMStreamingEndpoint_maxCacheAge(data),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(data.ResourceName, "max_cache_age_seconds", "60"),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func testCheckAzureRMStreamingEndpointDestroy(s *terraform.State) error {
+	conn := acceptance.AzureProvider.Meta().(*clients.Client).Media.StreamingEndpointsClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_media_streaming_endpoint" {
+			continue
+		}
+
+		id, err := parse.StreamingEndpointID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+		resp, err := conn.Get(ctx, id.ResourceGroup, id.MediaserviceName, id.Name)
+
+		if err != nil {
+			return nil
+		}
+
+		if resp.StatusCode != http.StatusNotFound {
+			return fmt.Errorf("Streaming Endpoint still exists:\n%#v", resp)
+		}
+	}
+
+	return nil
+}
+
+func testAccAzureRMStreamingEndpoint_basic(data acceptance.TestData) string {
+	template := testAccAzureRMStreamingEndpoint_template(data)
+	return fmt.Sprintf(`
+%s
+resource "azurerm_media_streaming_endpoint" "test" {
+  name                        = "endpoint1"
+  resource_group_name         = azurerm_resource_group.test.name
+  location                    = azurerm_resource_group.test.location
+  media_services_account_name = azurerm_media_services_account.test.name
+  scale_units                 = 1
+}
+`, template)
+}
+
+func testAccAzureRMStreamingEndpoint_CDN(data acceptance.TestData) string {
+	template := testAccAzureRMStreamingEndpoint_template(data)
+	return fmt.Sprintf(`
+%s
+resource "azurerm_media_streaming_endpoint" "test" {
+  name                        = "endpoint1"
+  resource_group_name         = azurerm_resource_group.test.name
+  location                    = azurerm_resource_group.test.location
+  media_services_account_name = azurerm_media_services_account.test.name
+  scale_units                 = 1
+  cdn_enabled                 = true
+  cdn_provider                = "StandardVerizon"
+  cdn_profile                 = "MyCDNProfile"
```

Comment:
presumably this needs to be the name of an actual CDN Profile?

Reply:
No, the profile is created with the streaming endpoint and it's named using this attribute